smart-contracts/advanced/deterministic-deployment.md

```diff
@@ -0,0 +1,141 @@
+## What is deterministic deployment?
+
+Deterministic deployment refers to the ability to deploy smart contracts to the same address across different blockchain networks. This consistency can be particularly useful when deploying the same contract on multiple chains to streamline interactions or ensure interoperability. There are two primary methods to achieve deterministic deployment:
+
+- Using a **factory contract that has been deployed keylessly** to deploy your contracts.
+    - The factory usually uses the `CREATE2` opcode to deploy your contracts, precomputing the deployment address and ensuring consistency across all chains.
+- **Deploying your contracts keylessly** (without using any factory) onto all blockchains.
+    - You must keep the code and constructor arguments unchanged to get the same address.
+
```

Comment:
Since the rest of the doc focuses on deploying factory contract or utilizing it, and NOT on deploying contracts from EOA, suggest you add a line at the end of first section, about how this article focuses on deploying factory contract and utilizing it.